src/strategies/api.ts

```diff
@@ -1,5 +1,34 @@
-// TODO: xhr call to private rest api goes here
+import InstagramClient from 'instagram-private-api';
+import { Strategy, LikeOptions } from "./strategy";
 
-export class RestApi {
+const Client = InstagramClient.V1;
 
+export class RestApi implements Strategy {
+  session?: any;
+
+  constructor() {
+
+  }
+
+  async login(userName: string, password: string): Promise<void> {
+    const device = new Client.Device(userName);
+    const storage = new Client.CookieFileStorage(`${__dirname}/${userName}.json`);
```

Comment:
This is actually cool, they have some cookie storage, so you don't need to login all the time